### PR TITLE
Fix web3rpc javascript codegen to handle optional parameters

### PR DESCRIPTION
**Pull request by blukat29**
- Fix `KlaytnJavascriptClientCodegen.kt` to inject `x-default-*` tags to mustache templates.
- Fix Javascript `api.mustache` to accept optional parameters as both Object and Array.
- Fix Javascript `api.mustache` to throw when "optional" parameters are not given.
- Fix js-ext-core to `promisifyMethod` pass "optional" parameters in.
- Eventually allow passing optional parameters from ethers-ext and web3js-ext providers.

Test:

```js
// ethers-ext/example/test.js
const { Wallet, JsonRpcProvider } = require("@klaytn/ethers-ext");
const { ethers } = require("ethers");

const { asyncOpenApi } = require("../../js-ext-core/dist/util/openapi.js");
const { KlayApi } = require("../../web3rpc/sdk/client/javascript/openapi/dist/index.js");

const senderAddr = "0xa2a8854b1802d8cd5de631e690817c253d6a9153";
const senderPriv = "0x0e4ca6d38096ad99324de0dde108587e5d7c600165ae4cd6c2462c597458c2b8";

const provider = new JsonRpcProvider("https://public-en-baobab.klaytn.net");
const wallet = new Wallet(senderPriv, provider);

// Verify a message signed by an AccountKeyLegacy account
async function verifyMsg() {

  const msg = "hello";
  const msghex = ethers.utils.hexlify(ethers.utils.toUtf8Bytes(msg));
  const sig = await wallet.signMessage(msg);
  console.log({ senderAddr, msg, msghex, sig });

  const addr1 = ethers.utils.verifyMessage(msg, sig);
  console.log("recoveredAddr lib", addr1, addr1.toLowerCase() === senderAddr);

  const addr2 = await provider.send("klay_recoverFromMessage", [senderAddr, msghex, sig, "latest"]);
  console.log("recoveredAddr rpc", addr2, addr2.toLowerCase() === senderAddr);

  // fails before PR
  // const addr3 = await provider.klay.recoverFromMessage(senderAddr, msghex, sig, "latest"); // fails before PR

  // succeeds after PR; quick test with un-released versions. Works with/without "latest"
  const send = (method, params) => provider.send(method, params);
  const klay = asyncOpenApi(send, KlayApi);
  // const addr3 = await klay.recoverFromMessage(senderAddr, msghex, sig, "latest");
  const addr3 = await klay.recoverFromMessage(senderAddr, msghex, sig);

  console.log("recoveredAddr rpc", addr3, addr3.toLowerCase() === senderAddr);
}

async function main() {
  await verifyMsg();
}
main().catch(console.error);
```

Generated file example (KlayApi.js):

```js
    key: "recoverFromMessage",
    value: function recoverFromMessage(address, message, signature, opts, callback) {
      opts = opts || {};
      // verify the required parameter 'address' is set
      if (address === undefined || address === null) {
        throw new Error("Missing the required parameter 'address' when calling recoverFromMessage");
      }
      // verify the required parameter 'message' is set
      if (message === undefined || message === null) {
        throw new Error("Missing the required parameter 'message' when calling recoverFromMessage");
      }
      // verify the required parameter 'signature' is set
      if (signature === undefined || signature === null) {
        throw new Error("Missing the required parameter 'signature' when calling recoverFromMessage");
      }
      var postBody = new _RecoverFromMessageRequest["default"](undefined, undefined, undefined);
      var optsIndex = 0;
      postBody.params = [];
      postBody.params.push(address);
      postBody.params.push(message);
      postBody.params.push(signature);
      var defaultValue = null;
      defaultValue = "latest";
      var optParam = opts['blockNumberOrTag'] || opts[optsIndex++] || defaultValue;
      if (optParam === undefined || optParam === null) {
        throw new Error("Missing the required parameter 'blockNumberOrTag' when calling recoverFromMessage");
      }
      postBody.params.push(optParam);
      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _RecoverFromMessage200Response["default"];
      return this.apiClient.callApi('/klay/recoverFromMessage', 'POST', null, null, null, null, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
```

### CODE REVIEW

File: web3rpc/sdk/client/javascript/template/libraries/javascript/api.mustache [hunk 1]
```diff
@@ -57,12 +67,16 @@ export default class <&classname> {
...
       let postBody = <#bodyParam>new <dataType>(undefined, undefined, undefined)</bodyParam>;
+      let optsIndex = 0;
       postBody.params = [];
       <#queryParams><#required>
       postBody.params.push(<paramName>);</required><^required>
-      if(opts['<paramName>']) {
-        postBody.params.push(opts['<paramName>']);
-      }</required></queryParams>
+      var optParam = opts['<paramName>'] || opts[optsIndex++];
+      if (optParam === undefined || optParam === null) {
```
Comment (kjeom) [inline web3rpc/sdk/client/javascript/template/libraries/javascript/api.mustache:75]: It's intended for users necessarily to enter the opts?

Comment (blukat29) [reply web3rpc/sdk/client/javascript/template/libraries/javascript/api.mustache:75]: Yes, it is intended. Currently, there seems no '<defualtValue>' passed to the template, so the user must supply them. If I don't throw it here, then the user gets a server RPC error which is less comprehensible.

We might fix 'codegen/src/main/kotlin/web3rpc/client/KlaytnJavascriptClientCodegen.kt' like Java did, but I did not.

Comment (blukat29) [reply web3rpc/sdk/client/javascript/template/libraries/javascript/api.mustache]: @kjeom Fixed the Kotlin code. PTAL.

**Review by kjeom — APPROVED** [review]

File: web3rpc/sdk/client/javascript/template/libraries/javascript/api.mustache (hunk 1 above)
Comment (kjeom) [reply web3rpc/sdk/client/javascript/template/libraries/javascript/api.mustache]: @blukat29 LGTM

**Pull request merged**